### ISSUE TITLE
forward() missing 1 required positional argument: 'input'

### ISSUE THREAD

**Issue by yuvrajkhanna** [pytorch/examples#438]
class Net(nn.Module):
    def __init__(self):
        super(Net,self).__init__()
        self.conv1=nn.Conv2d(3,10,kernel_size=3,padding=1)
        self.conv2=nn.Conv2d(10,20,kernel_size=3,padding=1)
        self.conv3=nn.Conv2d(20,40,kernel_size=3,padding=1)
        self.conv4=nn.Conv2d(40,80,kernel_size=3,padding=1)
        self.pool=nn.MaxPool2d(2)
        self.fc1=nn.Linear(1024,512)
        self.fc2=nn.Linear(512,10)
    def forward(self,x):
        x=F.relu(self.conv1(x))
        x=self.pool(F.relu(self.conv2(x)))
        x=F.dropout(x)
        x=F.relu(self.conv3(x))
        x=self.pool(F.relu(self.conv4(x)))
        x=x.view(-1,1024)
        x=self.fc1
        x=self.fc2
        return nn.LogSoftmax(x)

net=Net()
net=net.cuda()

for epoch in range(num_epochs):
    print(epoch)
    X_train,Y_train=shuffle(X_train,Y_train)
    for i in range(no_of_batches):
        start=i*batch_size
        end=(i+1)*batch_size
        X_var=Variable(torch.cuda.FloatTensor(X_train[start:end]))
        Y_var=Variable(torch.cuda.FloatTensor(Y_train[start:end]))
        optimizer.zero_grad()
        Y_pred=net(X_var)
        loss=criterion(Y_pred,Y_var)
        loss.backward()
        optimizer.step()

**Comment by msaroufim** [comment]
I would double check and print what `X_var` is before you pass it to `net()`
